patches/api/0175-Fix-Spigot-annotation-mistakes.patch

```diff
@@ -85,17 +85,43 @@ index f124b35ec76e6cb6a1a0dc464005087043c3efd0..94a2fef0dc9e13c754cd31d5eabc1bde
 +@Deprecated // Paper
  public interface LingeringPotion extends ThrownPotion { }
 diff --git a/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java b/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
-index 2ff1b1308571d8f8056d3359e8a8ba4a589c3726..e669ad8ecd182c6899c7820414e6ee1f7312d699 100644
+index 2ff1b1308571d8f8056d3359e8a8ba4a589c3726..d9acfaef353a5579864a7a3a5b8a876340cfc93b 100644
 --- a/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
 +++ b/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
-@@ -68,6 +68,7 @@ public class PrepareItemEnchantEvent extends InventoryEvent implements Cancellab
+@@ -9,6 +9,7 @@ import org.bukkit.event.inventory.InventoryEvent;
+ import org.bukkit.inventory.InventoryView;
+ import org.bukkit.inventory.ItemStack;
+ import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
+ 
+ /**
+  * Called when an ItemStack is inserted in an enchantment table - can be
+@@ -23,7 +24,7 @@ public class PrepareItemEnchantEvent extends InventoryEvent implements Cancellab
+     private boolean cancelled;
+     private final Player enchanter;
+ 
+-    public PrepareItemEnchantEvent(@NotNull final Player enchanter, @NotNull InventoryView view, @NotNull final Block table, @NotNull final ItemStack item, @NotNull final EnchantmentOffer[] offers, final int bonus) {
++    public PrepareItemEnchantEvent(@NotNull final Player enchanter, @NotNull InventoryView view, @NotNull final Block table, @NotNull final ItemStack item, @Nullable final EnchantmentOffer[] offers, final int bonus) { // Paper - offers can contain null values
+         super(view);
+         this.enchanter = enchanter;
+         this.table = table;
+@@ -68,6 +69,7 @@ public class PrepareItemEnchantEvent extends InventoryEvent implements Cancellab
       * @return experience level costs offered
       * @deprecated Use {@link #getOffers()} instead of this method
       */
 +    @Deprecated // Paper
      @NotNull
      public int[] getExpLevelCostsOffered() {
          int[] levelOffers = new int[offers.length];
+@@ -85,7 +87,7 @@ public class PrepareItemEnchantEvent extends InventoryEvent implements Cancellab
+      *
+      * @return list of available enchantment offers
+      */
+-    @NotNull
++    @Nullable // Paper - offers can contain null values
+     public EnchantmentOffer[] getOffers() {
```

Comment:
Well the correct signature is actually `@Nullable EnchantmentOffer @NotNull[] getOffers()`

Reply:
I am not used to nullability notations with Java as I normally use Kotlin, so i'll fix that.

Reply:
Fixed.